sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -334,6 +335,71 @@ class SQLContext(@transient val sparkContext: SparkContext)
     applySchema(rowRDD, appliedSchema)
   }
 
+  @Experimental
+  def load(
+    dataSourceName: String,
+    options: Map[String, String]): DataFrame = {
+    val resolved = ResolvedDataSource(this, None, dataSourceName, options)
+    new DataFrame(this, LogicalRelation(resolved.relation))
+  }
+
+  @Experimental
+  def load(
+    dataSourceName: String,
+    options: java.util.Map[String, String]): DataFrame = {
+    load(dataSourceName, options.toMap)
+  }
+
+  @Experimental
+  def loadAsTempTable(
```

Comment:
this seems redundant, since users can do

load(...).registerAsTempTable(...) ?